raven/src/main/java/com/getsentry/raven/DefaultRavenFactory.java

```diff
@@ -375,21 +377,25 @@ protected Marshaller createMarshaller(Dsn dsn) {
     }
 
     /**
-     * Provides a list of package names to consider as "not in-app".
+     * Returns the list of package names to consider "in-app".
      * <p>
-     * Those packages will be used with the {@link StackTraceInterface} to hide frames that aren't a part of
-     * the main application.
+     * Those packages will be used with the {@link StackTraceInterface} to show frames that are a part of
+     * the main application in the Sentry UI by default.
      *
-     * @return the list of "not in-app" packages.
-     */
-    protected Collection<String> getNotInAppFrames() {
-        return Arrays.asList("com.sun.",
-            "java.",
-            "javax.",
-            "org.omg.",
-            "sun.",
-            "junit.",
-            "com.intellij.rt.");
+     * @param dsn Sentry server DSN which may contain options.
+     * @return the list of package names to consider "in-app".
+     */
+    protected Collection<String> getInAppFrames(Dsn dsn) {
+        if (dsn.getOptions().containsKey(IN_APP_FRAMES_OPTION)) {
+            ArrayList<String> inAppPackages = new ArrayList<>();
+            for (String inAppPackage : dsn.getOptions().get(IN_APP_FRAMES_OPTION).split(",")) {
+                if (!inAppPackage.trim().equals("")) {
+                    inAppPackages.add(inAppPackage);
+                }
+            }
+            return inAppPackages;
+        }
+        return Collections.emptyList();
```

Comment:
Stylistic nitpick: usually, in this case, I'd invert the conditional at the top and return the empty value immediately at the top of the function, rather than wrapping the bulk of the "real" code in the conditional block. (Is this not idomatic Java style?)